make "build_all_ios.sh"  occur error

I try to build tensorflow support at Android and iOS by makefile [tutorial](https://github.com/tensorflow/tensorflow/tree/master/tensorflow/contrib/makefile
) in current master branch 04c318b69c5b565436cfeeaab1cb7fd5419dde27

When running the build_all_ios.sh script, the below error message show
```
Undefined symbols for architecture x86_64:
  "nsync::nsync_mu_init(nsync::nsync_mu_s_*)", referenced from:
      tensorflow::mutex::mutex() in env.o
      tensorflow::mutex::mutex() in random.o
  "nsync::nsync_mu_lock(nsync::nsync_mu_s_*)", referenced from:
      tensorflow::mutex::lock() in env.o
      tensorflow::mutex::lock() in random.o
      tensorflow::mutex::lock() in histogram.o
  "nsync::nsync_mu_unlock(nsync::nsync_mu_s_*)", referenced from:
      tensorflow::mutex::unlock() in env.o
      tensorflow::mutex::unlock() in random.o
      tensorflow::mutex::unlock() in histogram.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [/Users/CSL.Peter/tensorflow/tensorflow/tensorflow/contrib/makefile/gen/host_bin/proto_text] Error 1
+ '[' 2 -ne 0 ']'
+ echo 'armv7 compilation failed.'
armv7 compilation failed.
+ exit 1
```
The `download_dependencies.sh` and `compile_ios_protobuf.sh` run successfully but `compile_ios_tensorflow.sh` failed. I find same issues #3191 and #4252 and seem to be fixed at #4287, but this problem still happen.

I am getting the same error with tf1,3 on raspberry pi3
I did following 
----------------------
export HOST_NSYNC_LIB=`tensorflow/contrib/makefile/compile_nsync.sh`
export TARGET_NSYNC_LIB="$HOST_NSYNC_LIB"
make -f tensorflow/contrib/makefile/Makefile HOST_OS=PI TARGET=PI  OPTFLAGS="-Os -mfpu=neon-vfpv4 -funsafe-math-optimizations -ftree-vectorize" CXX=g++-4.8

sudo cp /home/pi/tensorflow/tensorflow/contrib/makefile/downloads/nsync/public/nsync* /usr/include/.
make -f tensorflow/contrib/pi_examples/label_image/Makefile
======== Now I start to get error-----

subprocess.cc:(.text+0xcd4): undefined reference to `nsync::nsync_mu_unlock(nsync::nsync_mu_s_*)'
collect2: error: ld returned 1 exit status

----------------------------

How do I address this error. I have tried all most all idea suggested on web .
I am running Jessie version 8 .

--- a/tensorflow/contrib/makefile/build_all_ios.sh
+++ b/tensorflow/contrib/makefile/build_all_ios.sh
@@ -96,7 +96,7 @@ if [[ "${ONLY_MAKE_TENSORFLOW}" != "true" ]]; then
 
     if [[ -z "${BUILD_ARCH}" ]]; then
         # Compile protobuf for the target iOS device architectures.
-        tensorflow/contrib/makefile/compile_ios_protobuf.sh -a ${DEFAULT_ARCH}
+        tensorflow/contrib/makefile/compile_ios_protobuf.sh
     else
         # Compile protobuf for the target iOS device architectures.
         tensorflow/contrib/makefile/compile_ios_protobuf.sh -a ${BUILD_ARCH}

